code/modules/awaymissions/corpse.dm

```diff
@@ -4,294 +4,474 @@
-		M.equip_to_slot_or_del(new corpsepocket1(M), slot_r_store)
-	if(corpsepocket2)
-		M.equip_to_slot_or_del(new corpsepocket2(M), slot_l_store)
-	if(corpseback)
-		M.equip_to_slot_or_del(new corpseback(M), slot_back)
-	if(corpseid == 1)
-		var/obj/item/weapon/card/id/W = new(M)
-		W.name = "[M.real_name]'s ID Card"
-		var/datum/job/jobdatum
-		for(var/jobtype in typesof(/datum/job))
-			var/datum/job/J = new jobtype
-			if(J.title == corpseidaccess)
-				jobdatum = J
-				break
-		if(corpseidicon)
-			W.icon_state = corpseidicon
-		if(corpseidaccess)
-			if(jobdatum)
-				W.access = jobdatum.get_access()
-			else
+		H.set_species(mob_species)
+	if(husk)
+		H.ChangeToHusk()
+	else //Because for some reason I can't track down, things are getting turned into husks even if husk = false. It's in some damage proc somewhere.
+		H.mutations.Remove(HUSK)
+	H.underwear = "Nude"
+	H.undershirt = "Nude"
+	H.socks = "Nude"
+	if(outfit)
+		var/static/list/slots = list("uniform", "r_hand", "l_hand", "suit", "shoes", "gloves", "ears", "glasses", "mask", "head", "belt", "r_pocket", "l_pocket", "back", "id", "neck", "backpack_contents", "suit_store")
+		for(var/slot in slots)
+			var/T = vars[slot]
+			if(!isnum(T))
+				outfit.vars[slot] = T
+		H.equipOutfit(outfit)
+		//if(disable_pda) //FETHAS FIX THIS
+			// We don't want corpse PDAs to show up in the messenger list.
+		//	var/obj/item/device/pda/PDA = locate(/obj/item/device/pda) in H
+		//	if(H.PDA)
+		//		H.PDA.toff = 1
+	var/obj/item/weapon/card/id/W = H.wear_id
+	if(W)
+		if(id_access)
+			for(var/jobtype in typesof(/datum/job))
+				var/datum/job/J = new jobtype
+				if(J.title == id_access)
+					W.access = J.get_access()
+					break
+		if(id_access_list)
+			if(!islist(W.access))
 				W.access = list()
-		if(corpseidjob)
-			W.assignment = corpseidjob
-		W.registered_name = M.real_name
-		M.equip_to_slot_or_del(W, slot_wear_id)
-	if(coffin == 1)
-		var/obj/structure/closet/coffin/sarcophagus/sarc = locate(/obj/structure/closet/coffin/sarcophagus) in loc
-		if(sarc) M.loc = sarc
+			W.access |= id_access_list
+		if(id_job)
+			W.assignment = id_job
+		W.registered_name = H.real_name
+		W.update_label()
+
+//Instant version - use when spawning corpses during runtime
+/obj/effect/mob_spawn/human/corpse
+	roundstart = FALSE
+	instant = TRUE
+
+/obj/effect/mob_spawn/human/corpse/damaged
+	brute_damage = 1000
+
+/obj/effect/mob_spawn/human/alive
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "sleeper"
+	death = FALSE
+	roundstart = FALSE //you could use these for alive fake humans on roundstart but this is more common scenario
+
+
+//Non-human spawners
+
+/obj/effect/mob_spawn/AICorpse/create() //Creates a corrupted AI
+	var/A = locate(/mob/living/silicon/ai) in loc
+	if(A)
+		return
+	var/mob/living/silicon/ai/M = new(loc) //spawn new AI at landmark as var M
+	M.name = src.name
+	M.real_name = src.name
+	//M.aiPDA.toff = 1 //turns the AI's PDA messenger off, stopping it showing up on player PDAs

```

Comment:
Commented code.

Reply:
its commeted out cuase i am not sure how to disable a PDA properly.